posts/distributed-tracing-with-xray.md

```diff
@@ -0,0 +1,269 @@
+---
+authors:
+- Amine Ouali Alami
+tags:
+- DevOps
+- Spring
+- Microservice
+- Monitoring
+- AWS
+- X-ray
+- Distributed tracing
+- Observability
+- Architecture
+date: 2021-02-01T20:16:10.000Z
+title: "Distributed Tracing using AWS X-Ray"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/02/Distributed-Tracing-using-AWS-X-Ray.png
+---
+
+As monoliths have been replaced by microservice applications, growing in size and complexity, 
+The challenge of troubleshooting and identifying performance issues increased.
+The extensive use of cloud platforms such as AWS, making these cloud architectures highly elastic and dynamic. Traditional tools to capture and analyse information become powerless.
+Architects, IT Ops analysts now require operational visibility into their entire architecture
+
+On AWS, we can use distributed tracing services, such as AWS X-Ray, to collect and record traces as requests travel through the application, generate maps to see how requests flow across services, gain insight on the relationships between components, identify, and analyze issues in real time.
```

Comment:
```suggestion
On AWS, we can use distributed tracing services, such as [AWS X-Ray](https://aws.amazon.com/xray/), to collect and record traces as requests travel through the application, generate maps to see how requests flow across services, gain insight on the relationships between components, as well as identify and analyze issues in real time.
```
Do we need to touch on the concept of what distributed tracing is here or is it assumed that the reader knows already?

Reply:
In the introduction, I described the problem that distributed tracing services are trying to solve.
In addition, the following sentence "to collect and record traces as requests travel through the application, generate maps to see how requests flow across services, gain insight on the relationships between components, as well as identify and analyze issues in real-time." acts as a definition.
Let me know if I need to make it needs more explanation.